src/main/java/com/epam/eco/commons/avro/converter/DefaultAvroConverters.java

```diff
@@ -339,16 +348,75 @@ public static class DefaultMapAvroConverter implements AvroConverter {
         public Object toAvro(Object value, Schema schema, AvroConverters converters) {
             if (value instanceof Map) {
                 Schema mapValuesSchema = schema.getValueType();
+    
+                MapValueNullableCollector mapValueNullableCollector = new MapValueNullableCollector(
```

Comment:
Adding dedicated Collector class seems excessive. Consider using simple for loop.